source/meshconv/meshconv.cpp

```diff
@@ -2,40 +2,56 @@
 #include "MRMesh/MRMeshLoad.h"
 #include "MRMesh/MRMeshSave.h"
 #include "MREAlgorithms/MREMeshDecimate.h"
+#include "MREAlgorithms/MREBooleanOperation.h"
 #include <boost/program_options.hpp>
 #include <boost/exception/diagnostic_information.hpp>
 #include <iostream>
 
+
 // can throw
 static int mainInternal( int argc, char **argv )
 {
     std::filesystem::path inFilePath;
+    std::filesystem::path inFilePathSecond;
     std::filesystem::path outFilePath;
     float targetEdgeLen = 0;
 
     namespace po = boost::program_options;
-    po::options_description desc( "Available options" );
-    desc.add_options()
+    po::options_description generalOptions( "General options" );
+    generalOptions.add_options()
         ("help", "produce help message")
         ("input-file", po::value<std::filesystem::path>( &inFilePath ), "filename of input mesh")
-        ("output-file", po::value<std::filesystem::path>( &outFilePath ), "filename of output mesh")
-        ("remesh", po::value<float>( &targetEdgeLen )->implicit_value( targetEdgeLen ), "optional argument if positive is target edge length after remeshing")
+        ("output-file,o", po::value<std::filesystem::path>( &outFilePath ), "filename of output mesh")
+        ;
+
+    po::options_description operations( "Operations" );
+    operations.add_options()
+        ( "remesh", po::value<float>( &targetEdgeLen )->implicit_value( targetEdgeLen ), "optional argument if positive is target edge length after remeshing" )
+        ( "unite", po::value<std::filesystem::path>( &inFilePathSecond ), "unite mesh from input file and given mesh" )
+        ( "subtract", po::value<std::filesystem::path>( &inFilePathSecond ), "subtract given mesh from input file mesh given mesh" )
+        ( "intersect", po::value<std::filesystem::path>( &inFilePathSecond ), "intersect mesh from input file and given mesh" )
         ;
 
+    po::options_description cmdLine( "Available options" );
+    cmdLine.add( generalOptions ).add( operations );
+
     po::positional_options_description p;
     p.add("input-file", 1);
     p.add("output-file", 1);
 
     po::variables_map vm;
-    po::store(po::command_line_parser(argc, argv).options(desc).positional(p).run(), vm);
+    po::store(po::command_line_parser(argc, argv).options( cmdLine ).positional(p).run(), vm);
     po::notify(vm);
 
-    if ( vm.count("help") || !vm.count("input-file") || !vm.count("output-file") )
+
+    if ( vm.count("help") || !vm.count("input-file") || !vm.count("output-file") ||
+        ( vm.count( "remesh" ) + vm.count( "unite" ) + vm.count( "subtract" ) + vm.count( "intersect" ) > 1) )
```

Comment:
let us allow many operations, just in some fixed order, and let remesh be the last

Reply:
reworked